### PR TITLE
Simplification with assumptions: refine and abs

### PR DESCRIPTION
**Pull request by maurogaravello**
This deals with  #12724. In case of `refine(abs(x*y), Q.positive(x))`
the result was `Abs(x*y)`, which is mathematically correct. A better result
is `x * Abs(y)`.

This PR proposes a change in `refine_abs()` method in `sympy/assumptions/refine.py`.
In the case the arg is a multiplication, the `Abs` is splitted according to the factors of arg.
A test is added in `sympy/assumptions/tests/test_refine.py`.

### CODE REVIEW

File: sympy/assumptions/refine.py [hunk 1]
```diff
@@ -70,6 +70,9 @@ def refine_abs(expr, assumptions):
         return arg
     if ask(Q.negative(arg), assumptions):
         return -arg
+    # arg is Mul
+    if isinstance(arg, Mul):
+        return refine(abs(arg.args[0]), assumptions) * refine(abs(arg / arg.args[0]), assumptions)
```
Comment (smichr) [inline sympy/assumptions/refine.py]: Might we keep those that don't simplify out of Abs *in* the Abs? e.g. `Abs(x, y, z) -> x * Abs(y*z)` if only x is positive?

```
If isinstance(arg, Mul):
    r = [refine(abs(a), assumptions) for a in arg.args]
    nonabs = []
    inabs = []
    for i in r:
        if isinstance(i, Abs):
            inabs.append(i.args[0])
        else:
            nonabs.append(i)
    return Mul(*nonabs)*Abs(Mul(*inabs))
```

Comment (maurogaravello) [inline sympy/assumptions/refine.py]: I agree with your  suggestion.
Probably today I will not able to push the commit.

**Comment by smichr** [comment]
Thanks -- it's in.

**Comment by maurogaravello** [comment]
Thanks

**Pull request merged**